### DISCUSSION TITLE
How can I ask for an article, instead of chat?

### DISCUSSION THREAD

**Discussion by talvasconcelos** [oobabooga/text-generation-webui#1809] [Q&A]
I'd like to ask a model (just like I can do on ChatGPT) to fill a blog post template, give an outline for the post/article and the write each chapter! 

I don't know where, and how, to "instruct" the model on this. I was using pygmalion-7b, and also erebus-7b but it seems that either i get a "chatty" response, or "I'll get on it" and then radio silence. 

Should i use some type of character for this? Instruct mode (which i can't quite figure out)? 

Also, sometime the model doesn't stop... keeps writing BS until running out of tokens I guess!

Any help for this?

**Comment by LaaZa** [comment]
Those models are not meant for instructions.
Pygmalion is for dialogue and roleplay.(nsfw)
Erebus is for nsfw novel writing.

Try models like Vicuna and StableVicuna. Choose the correct instruct template so they work the best.

**Reply by talvasconcelos** [reply]
So Erebus seems to be what I need! I want the AI to write NSFW stories/articles, no?

Can you tell me more about the instruct template?

**Reply by LaaZa** [reply]
Okay, but it is not finetuned for instruct format. So instead you will probably want to use either default or notebook UI modes. These show you the whole context and the model generates text directly after where the context ends. Erebus can understand `[Genre: cyberpunk, dystopia]` format to specify genres for it. Just add that at the top of the context. Then you can give it like a title you want it to generate text for. Just put the title there like it was written in a document or something and generate text after the title. You can also write a start for the text if you want to guide it into some specific way. Maybe you can write a synopsis before the story or something so it has reference what it should write about.

**Reply by LaaZa** [reply]
I want to also add that look out for the max token limit, which is 2048. When your text goes over this limit, the text at the beginning is not in the context, so it might not follow things up there anymore. Hopefully you have enough text in the story that helps it to stay on course.

**Reply by talvasconcelos** [reply]
Thank you so much for steering me into the right direction! I'll try it out.

**Reply by TomLucidor** [reply]
@LaaZa here are some questions
1. Are there any way of dealing with maximum token limits, and ways to expand them according to model limitations? Seen this article about "4x" one-liners https://kaiokendev.github.io/context
2. What are the top models for (NSFW) roleplay vs article writing vs instruction making?

**Reply by LaaZa** [reply]
> @LaaZa here are some questions
> 1. Are there any way of dealing with maximum token limits, and ways to expand them according to model limitations? Seen this article about "4x" one-liners https://kaiokendev.github.io/context
> 2. What are the top models for (NSFW) roleplay vs article writing vs instruction making?

Sorry but I do not wish to answer anymore on this repository since my work is not appreciated here.
However, I'll give you short answers this one time. And I'm happy to assist elsewhere.

1. Yes. Some newer models have employed methods that make them more flexible and can work beyond of their trained context length. RoPE is a method that can extend context at inference time, it is implemented in llama.cpp.

2. There is no short answer. You can ask from places like [TheBloke AI discord](https://discord.gg/theblokeai)
